### PR TITLE
Allow target to be a file and not only a dir

### PR DESCRIPTION
**Pull request by 0xDC0DE**
PR checklist:

- [ ] Documentation is up-to-date
- [ ] Changelog is up-to-date
- [ ] Change has no security implications (otherwise, ping security team)

### CODE REVIEW

**Comment by 0xDC0DE** [comment]
Fix for: https://github.com/returntocorp/semgrep/issues/4780

**Review by IagoAbal — APPROVED** [review]

**Comment by aryx** [comment]
Cc @underyx who has been recently working on consistency issues in the CLI

**Comment by minusworld** [comment]
@pieterdc1 Woohoo! 🚀 

I tried running this in your branch, but I think it might still need to filter to a single target file:

```
✗ semgrep --test --config use-shell-instruction.yaml use-shell-instruction.dockerfile
100%|████████|29/29 files
Found rule id mismatch - file=/Users/grayson/r2c/semgrep-rules/generic/dockerfile/best-practice/missing-no-install-recommends.dockerfile 'ruleid' annotation with no YAML rule={'missing-no-install-recommends'}
Found rule id mismatch - file=/Users/grayson/r2c/semgrep-rules/generic/dockerfile/best-practice/maintainer-is-deprecated.dockerfile 'ruleid' annotation with no YAML rule={'maintainer-is-deprecated'}
Found rule id mismatch - file=/Users/grayson/r2c/semgrep-rules/generic/dockerfile/best-practice/missing-image-version.dockerfile 'ruleid' annotation with no YAML rule={'missing-image-version'}
Found rule id mismatch - file=/Users/grayson/r2c/semgrep-rules/generic/dockerfile/best-practice/remove-package-cache.dockerfile 'ruleid' annotation with no YAML rule={'remove-package-cache'}
Found rule id mismatch - file=/Users/grayson/r2c/semgrep-rules/generic/dockerfile/best-practice/missing-apk-no-cache.dockerfile 'ruleid' annotation with no YAML rule={'missing-apk-no-cache'}
[...]
Found rule id mismatch - file=/Users/grayson/r2c/semgrep-rules/generic/dockerfile/best-practice/missing-dnf-assume-yes-switch.dockerfile 'ruleid' annotation with no YAML rule={'missing-dnf-assume-yes-switch'}
Found rule id mismatch - file=/Users/grayson/r2c/semgrep-rules/generic/dockerfile/best-practice/prefer-copy-over-add.dockerfile 'ruleid' annotation with no YAML rule={'prefer-copy-over-add'}
Failing due to rule id mismatch. There is a test denoted with 'ruleid: <rule name>' where the rule name does not exist or is not expected in the test file.
```

unless I invoked it incorrectly 🤔

**Comment by 0xDC0DE** [comment]
Oh! With my testing I had only used a directory with a single file. Default behaviour is that if the target is a file it will list all files in the target's parent directory. That does not work if we only select a single rule file, of course. I will fix that!

File: semgrep/semgrep/test.py [hunk 1]
```diff
@@ -302,8 +304,13 @@ def get_config_filenames(original_config: Path) -> List[Path]:
 def get_config_test_filenames(
     original_config: Path, configs: List[Path], original_target: Path
 ) -> Dict[Path, List[Path]]:
-    original_is_file_not_directory = original_target.is_file()
-    if original_is_file_not_directory:
+    original_config_is_file_not_directory = original_target.is_file()
```
Comment (IagoAbal) [inline semgrep/semgrep/test.py]:
```suggestion
    original_config_is_file_not_directory = original_config.is_file()
```

**Pull request merged**